File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> freq = new HashMap<>();

        for(int num: nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        PriorityQueue<Integer> pq = new PriorityQueue<>((n1, n2) ->  freq.get(n2) - freq.get(n1));

        for(int n: freq.keySet()) {
            pq.add(n);
        }

        int[] res = new int[k];

        for(int i=0; i<k; i++) {
            res[i] = pq.poll();
        }

        return res;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}

        for num in nums:
            if num not in freq:
                freq[num] = 0
            
            freq[num] += 1
        
        sorted_dict = dict(sorted(freq.items(), key=lambda x: x[1], reverse=True))

        res = []
        for i in sorted_dict:
            res.append(i)
            k -= 1
            if k == 0:
                break
        
        return res